EPOParser.py

```diff
@@ -4,115 +4,133 @@
+    return functools.reduce(
+        lambda acc, val: (acc + [val]) if predicateFn(val) else acc, coll, [])
+
 
 def fcmap(fn):
-  return lambda coll: functools.reduce(lambda acc, val: acc + [fn(val)], coll, [])
+    return lambda coll: functools.reduce(
+        lambda acc, val: acc + [fn(val)], coll, [])
+
 
 def fcfilter(predicateFn):
-  return lambda coll: functools.reduce(lambda acc, val: (acc + [val]) if predicateFn(val) else acc, coll, [])
+    return lambda coll: functools.reduce(
+        lambda acc, val: (acc + [val]) if predicateFn(val) else acc, coll, [])
+
 
 def fzip(coll1, coll2):
-  return list(zip(coll1, coll2))
+    return list(zip(coll1, coll2))
+
 
 def fcompose(*functions):
-  return functools.reduce(lambda f, g: lambda x: f(g(x)), functions, lambda x: x)  
+    return functools.reduce(
+        lambda f, g: lambda x: f(g(x)), functions, lambda x: x)
 
 ###########
-# helpers # 	
+# helpers #
 ###########
 
 def marker(id):
-  return id + '-parsecomplete'
+    return id + '-parsecomplete'
 
-# fromElement :: attrib | text -> Element -> {} | String
-def fromElement(info):
-  return lambda el: getattr(el, info)
+# from_element :: attrib | text -> Element -> {} | String
+def from_element(info):
+    """This is a docstring, which is used to comment on a function."""
+    return lambda el: getattr(el, info)
 
 # tree :: String -> Tree
 def tree(filename):
-  return etree.parse(filename, etree.XMLParser(encoding='utf-8', recover=True))
-
-# getFileInfo :: String -> [(filename, file-location)]
-def getFileInfo(indexFilename):
-  return fzip(
-    fmap(fromElement('text'), getElement(tree(indexFilename), '//filename')),
-    fmap(fromElement('text'), getElement(tree(indexFilename), '//file-location'))
-  )
-
-# getElement :: Tree, xpath -> [Element]
-def getElement(tree, elementPath):
-  return tree.xpath(elementPath)
-
-# getMetadata :: Tree, xpath, attrib | text, String -> [] | [(alias, tag, {} | String)]
-def getMetadata(tree, elementPath, info, alias):
-  return fmap(lambda el: ( alias, el.tag, getattr(el, info) ), getElement(tree, elementPath))
+    """
+    This is a multi-line docstring.
+
+    It is used when there is more info that should be shared than fits on one
+    line.
+    """
+    return etree.parse(
+        filename, etree.XMLParser(encoding='utf-8', recover=True))
+
+# get_file_info :: String -> [(filename, file-location)]
+def get_file_info(indexFilename):
+    return fzip(
+        fmap(from_element('text'), get_element(tree(indexFilename), '//filename')),
+        fmap(from_element('text'), get_element(tree(indexFilename), '//file-location'))
+    )
+
+# get_element :: Tree, xpath -> [Element]
+def get_element(tree, elementPath):
+    return tree.xpath(elementPath)
+
+# get_metadata :: Tree, xpath, attrib | text, String -> [] | [(alias, tag, {} | String)]
+def get_metadata(tree, elementPath, info, alias):
+    return fmap(lambda el: ( alias, el.tag, getattr(el, info) ), get_element(tree, elementPath))
```

Comment:
Following up on my last comment, you could replace `get_element(tree, element_path)` with `tree.xpath(element_path)`. Now we don't need the helper function.